File: components/engine/src/test/java/org/storm/engine/example/EntityImpl.java
package org.storm.engine.example;

import javafx.scene.paint.Color;
import org.storm.core.render.Renderable;
import org.storm.physics.entity.Entity;
import org.storm.physics.math.geometry.shapes.Shape;
import org.storm.physics.transforms.UnitConvertor;

public class EntityImpl extends Entity {

  public EntityImpl(Shape hurtBox, double speed, double mass, double restitution) {
    super(hurtBox, speed, mass, restitution);
  }

  @Override
  public Renderable transform(UnitConvertor unitConvertor) {
    return (gc, x, y) -> {
      gc.setFill(Color.BLACK);
      super.transform(unitConvertor).render(gc, x, y);
    };
  }

}

File: components/engine/src/test/kotlin/org/storm/engine/example/EntityImpl.kt
package org.storm.engine.example

import org.storm.core.graphics.canvas.Canvas
import org.storm.physics.entity.Entity
import org.storm.physics.math.geometry.shapes.Shape

class EntityImpl(
    hurtBox: Shape,
    speed: Double,
    mass: Double,
    restitution: Double
) : Entity(
    hurtBox,
    speed,
    mass,
    restitution
) {

    override suspend fun render(canvas: Canvas, x: Double, y: Double) {
        super.render(canvas, x, y)
    }

}
